recipe/meta.yaml

```diff
@@ -11,29 +12,60 @@ source:
     - have_openssl.patch
 
 build:
-  number: 2
-
-requirements:
-  build:
-    - python                                 # [build_platform != target_platform]
-    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
-    - postgresql                             # [build_platform != target_platform]
-    - {{ compiler('c') }}
-  host:
-    - python
-    - pip
-    - openssl  # [not linux]
-    - postgresql
-  run:
-    - python
-    # libpq is a repackaging of only the library to connect to postgres
-    - libpq
-    - openssl  # [not linux]
-
-test:
-  imports:
-    - psycopg2
-    - psycopg2._psycopg
+  number: 3
+
+outputs:
+  - name: {{ name }}
+    script: install_psycopg2.sh  # [unix]
+    script: install_psycopg2.bat  # [win]
+    requirements:
+      build:
+        - python                                 # [build_platform != target_platform]
+        - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+        - postgresql                             # [build_platform != target_platform]
+        - {{ compiler('c') }}
+      host:
+        - python
+        - pip
+        - openssl  # [not linux]
+        - postgresql
+      run:
+        - python
+        # libpq is a repackaging of only the library to connect to postgres
+        - libpq
+        - openssl  # [not linux]
+
+    test:
+      imports:
+        - psycopg2
+        - psycopg2._psycopg
+      commands:
+        - pip check
+      requires:
+        - pip
+
+  # alternative name for the package, patched to rename the package in the
+  # metadata so pip check will pass for packages that depend on pyscopg2-binary
+  - name: psycopg2-binary
+    script: install_psycopg2-binary.sh  # [unix]
+    script: install_psycopg2-binary.bat  # [win]
+    requirements:
+      build:
+        - python                                 # [build_platform != target_platform]
+        - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+      host:
+        - python
+        - pip
+      run:
+        - python
+        - {{ pin_subpackage(name, max_pin='x.x.x.x.x.x') }}
+    test:
+      imports:
+        - psycopg2
+      commands:
+        - pip check
```

Comment:
This looks great... can we try to import _something_ that current fails, and test against that?

Reply:
I don't see any clear way to add something here that *depends* on this feedstock.  That would require building the downstream package here with the new `psycop2-binary` dependency.   If you can come up with a more concrete idea of how to test here, I'm open to the idea but short of adding *another* metapackage here, this time one that depends on `psycop2-binary`, I can't see how to do this.

Reply:
Anything that, today:
- **works**, even though it says it "needs" `psycopg2-binary`
- fails `pip check`

said candidate's need for `psycopg2` should be satisfied, and this new little gem will take care of the pip check. Unfortunately, the ones i have personally been bit by are are monsters, like dagster and airflow...

Reply:
So the problem is that downstream packages will have to switch from depending on `psycopg2` to `psycopg2-binary` before `pip check` will succeed.  So I can't just add one of them here for testing because it would still depend on the `psycopg2` conda package and `pip check` would still fail.

Reply:
I think the latest `airflow` doesn't depend on `psycopg2` directly, just via one of its providers (`apache-airflow-providers-postgres`), which is what I already used in my local testing.

Reply:
right, but as `psycopg2` is now a dependency of `psycopg2-binary`, _conda_ will be happy, but we can demonstrate that `super-long-package-name` would _also_ be `pip check`able. We don't need to keep it in, because circular dependencies, but just once as brown M&Ms...

Reply:
There are currently no packages to check this on because `psycopg2-binary` doesn't exist on conda-forge.  So I still don't see how I can check this.

Reply:
In my local test, I'm calling `pip check`, so I've been able to demonstrate with a local build that `pip check` is happy on `apache-airflow-providers-postgres`.